salt/modules/mount.py

```diff
@@ -934,7 +942,7 @@ def set_automaster(
                 try:
                     with salt.utils.files.fopen(config, 'w+') as ofile:
                         # The line was changed, commit it!
-                        ofile.writelines(lines)
+                        ofile.writelines(salt.utils.data.decode(lines))
```

Comment:
Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.